lib/google_one_tap.rb

```diff
@@ -0,0 +1,117 @@
+# frozen_string_literal: true
+
+module OmniAuth
+  module Strategies
+    class GoogleOneTap
+      class GoogleOneTapValidationError < StandardError; end
+      class GoogleOneTapCSFRError < StandardError; end
+      include OmniAuth::Strategy
+
+      option :name , "google_one_tap"
+      ALLOWED_ISSUERS = ['accounts.google.com', 'https://accounts.google.com'].freeze
+
+      # This is wbhere you pass the options you would pass when
+      # initializing your consumer from the OAuth gem.
+      attr_accessor :access_token
+      option :provider_ignores_state, true
+
+      # We don't need this phase, Google one tap/idenentity framework dones't require it
+      # This is why this plugin doesn't need client_secret. as well
+      def request_phase
+        raise NotImplementedError
+      end
+
+      # These are called after authentication has succeeded (From Google side!).
+      def callback_phase
+        # We first check for CSRF and then validate the token.
+        # Ref: https://developers.google.com/identity/gsi/web/guides/verify-google-id-token
+        if check_csrf
+          self.access_token = build_access_token # It builds and validate it!.
+        end
+        super
+      end
+      def transformTokenToOmniauth(payload, credential)
+        auth_token = {
+          provider: "google_one_tap",
+          info: {
+            name: payload["name"],
+            email: payload["email"],
+            email_verified: payload["email_verified"], # This always true (Accordoing to Google)
```

Comment:
Typo
```suggestion
            email_verified: payload["email_verified"], # This always true (According to Google)
```
Also, do you have a reference for that information which can be linked in the comment?

Reply:
I was just rechecking the doc, I don't think my comment is totally accurate. 
While I couldn't find that explicitly in the [official reference](https://developers.google.com/identity/gsi/web/reference/js-reference#credential). I think its going to be `true` if their email suffix is `@gmail`.